File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.cpp
#include <string>
#include <stack>

using namespace std;

int solution(string s) {
    stack<char> st;

    for (char ch : s) {
        if (!st.empty() && st.top() == ch) {
            st.pop();
        } else {
            st.push(ch);
        }
    }

    return st.empty() ? 1 : 0;
}

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.java
import java.util.ArrayDeque;

class Solution {
    public int solution(String s) {
        ArrayDeque<Character> stack = new ArrayDeque<>();

        for (char c : s.toCharArray()) {
            if (!stack.isEmpty() && stack.peek() == c) {
                stack.pop();
            } else {
                stack.push(c);
            }
        }

        return stack.isEmpty() ? 1 : 0;
    }
}